src/error/Error.h

```diff
@@ -11,29 +11,53 @@
 
 #include "error/Detail.h"
 #include "error/forward.h"
+#include "mem/PoolingAllocator.h"
 
 #include <iosfwd>
+#include <vector>
+
+/// zero or more details of a single error
+using ErrorDetails = std::vector<ErrorDetailPointer, PoolingAllocator<ErrorDetailPointer> >;
+
+/// prints all given details separated by '+';
+/// if no details were given, prints nothing
+std::ostream &operator <<(std::ostream &, const ErrorDetails &);
 
 /// a transaction problem
 class Error {
 public:
     Error() = default;
     Error(const err_type c): category(c) {} ///< support implicit conversions
-    Error(const err_type c, const ErrorDetailPointer &d): category(c), detail(d) {}
+    Error(const err_type c, const ErrorDetailPointer &d): Error(c) { update(d); }
 
     explicit operator bool() const { return category != ERR_NONE; }
 
     /// if necessary, stores the given error information (if any)
     void update(const Error &);
 
-    /// convenience wrapper for update(Error)
-    void update(const err_type c, const ErrorDetailPointer &d) { update(Error(c, d)); }
```

Comment:
We could keep the old simple implementation, but it becomes quite expensive when we migrate from a single error detail to many details (regardless of the STL container type).